src/main/java/gregtech/api/fluids/MetaFluids.java

```diff
@@ -0,0 +1,253 @@
+package gregtech.api.fluids;
+
+import gregtech.api.GTValues;
+import gregtech.api.GregTechAPI;
+import gregtech.api.fluids.fluidType.FluidType;
+import gregtech.api.fluids.fluidType.FluidTypes;
+import gregtech.api.unification.material.Material;
+import gregtech.api.unification.material.Materials;
+import gregtech.api.unification.material.info.MaterialIconSet;
+import gregtech.api.unification.material.info.MaterialIconType;
+import gregtech.api.unification.material.properties.FluidProperty;
+import gregtech.api.unification.material.properties.PlasmaProperty;
+import gregtech.api.unification.material.properties.PropertyKey;
+import gregtech.api.util.FluidTooltipUtil;
+import gregtech.api.util.GTUtility;
+import gregtech.api.util.LocalizationUtils;
+import gregtech.common.blocks.MetaBlocks;
+import net.minecraft.client.renderer.texture.TextureMap;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.text.TextFormatting;
+import net.minecraftforge.fluids.BlockFluidBase;
+import net.minecraftforge.fluids.BlockFluidClassic;
+import net.minecraftforge.fluids.Fluid;
+import net.minecraftforge.fluids.FluidRegistry;
+import org.apache.commons.lang3.tuple.Pair;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.*;
+
+public class MetaFluids {
+
+    private static final Set<ResourceLocation> fluidSprites = new HashSet<>();
+    private static final Map<String, Material> fluidToMaterialMappings = new HashMap<>();
+    private static final Map<String, String> alternativeFluidNames = new HashMap<>();
+    private static final Map<Pair<Material, FluidType>, ResourceLocation> fluidTextureMap = new HashMap<>();
```

Comment:
Would prefer to use `Material` as key then have a `Map<FluidType, ResourceLocation>` as value, and again `Object2ObjectOpenHashMap`